website/docs/r/appsync_resolver.html.markdown

```diff
@@ -108,8 +108,8 @@ The following arguments are supported:
 * `api_id` - (Required) The API ID for the GraphQL API.
 * `type` - (Required) The type name from the schema defined in the GraphQL API.
 * `field` - (Required) The field name from the schema defined in the GraphQL API.
-* `request_template` - (Required) The request mapping template for UNIT resolver or 'before mapping template' for PIPELINE resolver.
-* `response_template` - (Required) The response mapping template for UNIT resolver or 'after mapping template' for PIPELINE resolver.
+* `request_template` - (Optional) The request mapping template for UNIT resolver or 'before mapping template' for PIPELINE resolver.
+* `response_template` - (Optional) The response mapping template for UNIT resolver or 'after mapping template' for PIPELINE resolver.
```

Comment:
It actually might be worth stating that these are only really option if `type = "AWS_LAMBDA"` (although there's no code-level check for that atm -- not sure if needed since it should just fail on AWS side during `apply`, but nice if we want feedback within terraform). Also might be nice to add an example to the code blocks on this page for "Direct to lambda resolver" or something.

Reply:
I agree here - we should probably note the requirement for non-Lambda resolvers. 👍